components/brave_sync/brave_profile_sync_service_impl.cc

```diff
@@ -280,6 +357,14 @@ void BraveProfileSyncServiceImpl::OnSetupSyncNewToSync(
 
   brave_sync_initializing_ = true;
 
+  RunWhenModelIsLoaded(
```

Comment:
I'm concerned about running parts of these methods, but not completing them. I think it's likely to introduce subtle bugs. I think it would be better to just wait on everything so we'd have `OnSetupSyncNewToSync` that would call `OnSetupSyncNewToSyncImpl` when the bookmark model is loaded
```

Reply:
thanks, now this is redone at https://github.com/brave/brave-core/pull/4588/commits/361b930cb4df1405b5f89d10e5673bd061f7b7db